data/mods/ssb/moves.ts

```diff
@@ -466,6 +466,39 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
 		type: "Normal",
 	},
 
+	// DragonWhale
+	cloakdance: {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "If Mimikyu's Disguise is intact or if Mimikyu is the last remaining Pokemon, Attack goes up 2 stages. If Mimikyu's Disguise is busted and there are other Pokemon on Mimikyu's side, the Disguise will be repaired and Mimikyu will switch out.",
+		shortDesc: "If Disguise intact or Mimikyu is the last remaining Pokemon, raises attack by 2; otherwise repairs Disguise and switches out.",
+		name: "Cloak Dance",
+		pp: 5,
+		priority: 0,
+		flags: {dance: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, ['Teleport', 'Swords Dance'][this.random(2)], target);
```

Comment:
Uh, i dont think thats what dragonwhale intended as the animation. I think its dependant on whether disguise breaks or not.